src/fmu/ensemble/ensemble.py

```diff
@@ -295,12 +278,12 @@ def add_realizations(self, paths, realidxregexp=None, autodiscovery=True):
         if USE_CONCURRENT:
             args = [(realdir, realidxregexp, autodiscovery) for realdir in globbedpaths]
             with ProcessPoolExecutor() as executor:
-                for realdir, realization in executor.map(
-                    self._init_scratch_realization, *zip(*args)
-                ):
-                    count += self._check_loading_of_realization(
-                        realization, realdir, realidxregexp
-                    )
+                realfutures = [executor.submit(ScratchRealization, realdir, realidxregexp=realidxregexp,
+                    autodiscovery=autodiscovery) for realdir in globbedpaths]
+
+            for realfuture in realfutures:
+                count += self._check_loading_of_realization(
+                    realfuture.result(), "dummydirfornow", None)
```

Comment:
Why did you use dummy arguments?

Reply:
I updated those lines in the meanwhile by the way - let me know if that was not what you had in mind.

And I see that for the other usage there is not this additional wrapper (there is one, but it is different.). Though, we might consider changing that to the `submit()` method as well - just to use the same thing.

Reply:
dummydir was just for laziness. Need to find a pretty solution, but it is only used for logging.